src/__test__/components/data-exploration/heatmap/HeatmapPlot.test.jsx

```diff
@@ -1,545 +1,342 @@
-      cellSets: {
-        ...initialState.cellSets,
-        hierarchy: [],
-        properties: [],
-        loading: false,
-        error: true,
-      },
-    });
+  it('Handles marker genes loading error correctly', async () => {
+    fetchMock.mockIf(/.*/, mockAPI(generateDefaultMockAPIResponses(experimentId, fake.PROJECT_ID)));
 
-    component = mount(
-      <Provider store={store}>
-        <HeatmapPlot experimentId={experimentId} width={200} height={200} />
-      </Provider>,
+    const customWorkerResponses = _.merge(
+      _.cloneDeep(mockWorkerResponses),
+      { '5-marker-genes': errorResponse },
     );
 
-    expect(component.find('HeatmapPlot').length).toEqual(1);
-    expect(store.getActions().length).toEqual(1);
-    const [loadAction] = store.getActions();
+    seekFromAPI.mockImplementation((a, b, c, requested) => customWorkerResponses[requested]());
+
+    await loadAndRenderDefaultHeatmap();
 
-    expect(loadAction.type).toBe(CELL_SETS_LOADING);
-    expect(loadAction).toMatchSnapshot();
+    // Error screen shows up
+    expect(screen.getByText(/We're sorry, we couldn't load this./i)).toBeInTheDocument();
   });
 
-  it('loads marker genes if it can', async () => {
-    const store = mockStore({
-      ...initialState,
-      networkResources: { environment: Environment.DEVELOPMENT },
-      cellSets: {
-        ...initialState.cellSets,
-        loading: true,
-        error: false,
-      },
-      genes: {
-        ...initialState.genes,
-        expression: {
-          ...initialState.genes.expression,
-          loading: false,
-          error: false,
-        },
-      },
-      experimentSettings: {
-        ...initialState.experimentSettings,
-        processing: {
-          ...initialState.processing,
-          configureEmbedding: {
-            clusteringSettings: { methodSettings: { louvain: { resolution: 10 } } },
-          },
-        },
-      },
-    });
+  it('Handles expression data loading error correctly', async () => {
+    fetchMock.mockIf(/.*/, mockAPI(generateDefaultMockAPIResponses(experimentId, fake.PROJECT_ID)));
 
-    component = mount(
-      <Provider store={store}>
-        <HeatmapPlot experimentId={experimentId} width={200} height={200} />
-      </Provider>,
+    const customWorkerResponses = _.merge(
+      _.cloneDeep(mockWorkerResponses),
+      { 'loading_gene_id-expression': errorResponse },
     );
 
-    expect(component.find('HeatmapPlot').length).toEqual(1);
+    seekFromAPI.mockImplementation((a, b, c, requested) => customWorkerResponses[requested]());
+
+    await loadAndRenderDefaultHeatmap();
+
+    // Renders correctly
+    expect(screen.getByText(/Sup Im a heatmap/i)).toBeInTheDocument();
 
-    const expectedMarkerGenes = 5;
+    // If a new gene suddenly is suddenly selected
+    await act(async () => {
+      storeState.dispatch(loadGeneExpression(experimentId, [...markerGenesData5.order, 'loading_gene_id'], 'interactiveHeatmap'));
+    });
```

Comment:
why would this result in an error?

Reply:
Because I set it so that the api returns an error here:

```
const customWorkerResponses = _.merge(
      _.cloneDeep(mockWorkerResponses),
      { 'loading_gene_id-expression': errorResponse },
    );
```